src/audio/copier/copier.c

```diff
@@ -1689,13 +1695,18 @@ static int copier_get_attribute(struct comp_dev *dev, uint32_t type, void *value
 static int copier_position(struct comp_dev *dev, struct sof_ipc_stream_posn *posn)
 {
 	struct copier_data *cd = comp_get_drvdata(dev);
+	int ret = 0;
 
 	/* Exit if no endpoints */
 	if (!cd->endpoint_num)
 		return -EINVAL;
 
+	if (dev->ipc_config.type == SOF_COMP_HOST && !cd->ipc_gtw)
+		posn->host_posn = cd->hd->local_pos;
+	else
+		ret = comp_position(cd->endpoint[IPC4_COPIER_GATEWAY_PIN], posn);
 	/* Return position from the default gateway pin */
-	return comp_position(cd->endpoint[IPC4_COPIER_GATEWAY_PIN], posn);
+	return ret;
```

Comment:
is this correct? you will return 0 in the case of COMP_HOST. Shouldn't you return posn->host_posn?

Reply:
sorry, changed.